module-utils/time/test/unittest_time.cpp

```diff
@@ -519,6 +519,12 @@ TEST_CASE("Duration - creation")
         REQUIRE(duration.get() == 333);
     }
 
+    SECTION("std::chrono::seconds constructor")
+    {
+        utils::time::Duration duration(std::chrono::seconds(333));
+        REQUIRE(duration.get() == 333);
```

Comment:
what will happen, when you pass e.g. minutes? duration will be presented in minutes, or casted to seconds?
I mean - should we return the count, or absolute duration?

Reply:
As Duration now takes second I would suppouse that minues will be implictly converted by chrono library

Reply:
I won't agree there - looking what you wrote, duration will be feeded with argument's `count` - so it will have count only, not absolute duration in seconds. If that's desired (that we have duration in seconds) you should modify the `get` method and add casting to seconds there

Reply:
that's why both good and bad path in unit tests are useful ;)

Reply:
test like this would pass:
 SECTION("std::chrono::seconds by minutes constructor")
    {
        auto minutes = std::chrono::minutes(2);
        utils::time::Duration duration(minutes);
        REQUIRE(duration.get() == 120);
    }
    Could  you please  give me real example how to break it ?

Reply:
my bad - I didn't see that you made the constructor requires `std::chrono::seconds` instead of generic duration: https://en.cppreference.com/w/cpp/chrono/duration

So the casting/conversion is going under the hood.